### PR TITLE
Fix bug where callable functions can't be emulated.

### PR DESCRIPTION
**Pull request by taeold**
Fixes https://github.com/firebase/firebase-tools/issues/4313

In https://github.com/firebase/firebase-tools/pull/4310, we started to support new type of triggers `callableTrigger` (distinct from `httpTrigger`).

Functions Emulator does not support `callableTrigger`. We fix this by converting `callbleTrigger` to `httpTrigger`. This is tech debt - at some point, we should use `endpoint` definitions throughout the emulator (cc @joehan).

Also adds integration tests for emulating callable triggers to prevent this kind of breakages in the future.

### CODE REVIEW

File: scripts/integration-helpers/framework.ts [hunk 1]
```diff
@@ -263,6 +263,21 @@ export class TriggerEndToEndTest {
     return fetch(url);
   }
 
+  invokeCallableFunction(
+    name: string,
+    body: Record<string, unknown>,
+    zone = FIREBASE_PROJECT_ZONE
+  ): Promise<Response> {
+    const url = `http://localhost:${[this.functionsEmulatorPort, this.project, zone, name].join(
```
Comment (bkendall) [inline scripts/integration-helpers/framework.ts:271]: This is real strange, just given how URLs should be thought of. Consider pulling the port out of the path:

```suggestion
    const url = `http://localhost:${this.functionsEmulatorPort}/${this.project, zone, name].join(
```

File: scripts/triggers-end-to-end-tests/functions/index.js [hunk 2]
```diff
@@ -359,3 +364,8 @@ exports.storagebucketv2metadatareaction = functionsV2.storage.onObjectMetadataUp
     return true;
   }
 );
+
+exports.oncallv2 = functionsV2.https.onCall((req) => {
```
Comment (bkendall) [inline scripts/triggers-end-to-end-tests/functions/index.js:368]: `onCallv2` Doesn't match `onCall` above as written

**Review by bkendall — CHANGES_REQUESTED** [review]

File: scripts/triggers-end-to-end-tests/functions/index.js (hunk 2 above)
Comment (taeold) [reply scripts/triggers-end-to-end-tests/functions/index.js:368]: v2 functions can't contain capital letters.

**Review by bkendall — APPROVED** [review]

**Review by colerogers — APPROVED** [review]

**Comment by BaptisteCool** [comment]
Release in 1.4.0 google [here](https://github.com/firebase/firebase-tools/issues/4320)

**Pull request merged**